### ISSUE TITLE
Failed to reconfigure connector's tasks, retrying after backoff

### ISSUE THREAD

**Issue by gregroberti** [jcustenborder/kafka-connect-spooldir#12]
**Stacktrace**
[2016-11-09 18:31:17,052] ERROR Failed to reconfigure connector's tasks, retrying after backoff: (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
java.lang.NullPointerException
        at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:51)
        at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:62)
        at org.apache.kafka.connect.runtime.ConnectorConfig.<init>(ConnectorConfig.java:79)
        at org.apache.kafka.connect.runtime.SourceConnectorConfig.<init>(SourceConnectorConfig.java:29)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.reconfigureConnector(DistributedHerder.java:845)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.reconfigureConnectorTasksWithRetry(DistributedHerder.java:799)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.startConnector(DistributedHerder.java:795)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.startWork(DistributedHerder.java:755)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.handleRebalanceCompleted(DistributedHerder.java:715)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.tick(DistributedHerder.java:206)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.run(DistributedHerder.java:176)
        at java.lang.Thread.run(Thread.java:745)

**Configuration used:**
[2016-11-09 18:31:16,983] INFO SpoolDirectoryConfig values:
        processing.file.extension = .PROCESSING
        csv.first.row.as.header = true
        finished.path = /tmp/spooldir/finished
        csv.strict.quotes = false
        csv.schema.name = BlockingIA
        csv.verify.reader = true
        csv.schema.from.header = true
        csv.file.charset = UTF-8
        file.minimum.age.ms = 0
        record.processor.class = class io.confluent.kafka.connect.source.io.processing.csv.CSVRecordProcessor
        csv.skip.lines = 0
        input.file.pattern = ^.*\.csv$
        error.path = /tmp/spooldir/error
        csv.schema.from.header.keys = [DayID, XXX, YYY]
        csv.quote.char = 34
        include.file.metadata = false
        csv.parser.timestamp.date.formats = [yyyy-MM-dd' 'HH:mm:ss]
        csv.schema =
        batch.size = 1000
        csv.parser.timestamp.timezone = UTC
        csv.null.field.indicator = BOTH
        halt.on.error = false
        csv.ignore.quotations = false
        csv.escape.char = 92
        csv.case.sensitive.field.names = false
        csv.ignore.leading.whitespace = true
        csv.keep.carriage.return = false
        csv.separator.char = 124
        topic = quickstart-data
        input.path = /tmp/spooldir/input
 (io.confluent.kafka.connect.source.SpoolDirectoryConfig)

**Comment by gregroberti** [comment]
I was able to resolve this issue by removing the 'name' from the configuration when sending the POST call as well as including guava-18.0.jar in the classpath.

**Comment by jcustenborder** [comment]
I'll take a look at this. I typically use [schema-registry](https://github.com/confluentinc/schema-registry) which most likely pulls in guava. Thanks for the bug report!

**Comment by jcustenborder** [comment]
Are you using this in your config?
```
key.converter=org.apache.kafka.connect.json.JsonConverter
value.converter=org.apache.kafka.connect.json.JsonConverter
```

**Comment by gregroberti** [comment]
Yes, I'm starting up kafka-connect using these values.

**Comment by jcustenborder** [comment]
Ok that makes sense! Thanks!

**Comment by cameron** [comment]
@jcustenborder, I'm 100% abusing your open source presence here to ask a selfish question, so feel free to ignore me: why does it make sense that the JsonConverter would be connected to the OP's error? I'm getting that error in one connector (debezium's, not yours), and not in another, and am super stumped as to why, b/c they're created with the same script and only a few minor config values vary between them (db, name, bootstrap servers...)...

**Comment by jcustenborder** [comment]
This one is really old and I hope I remember the context fully. I think this was in the early days before connect had support for plugin path isolation. During these times you would just copy the plugin and it's dependencies to the classpath of the connect worker. This made maintaining connectors very difficult and if I remember correctly *I did not ship guava or jackson* and tried to use the version that AK shipped. I work for Confluent so I tend to test using the packages that Confluent provides. The classpath for Confluents build of Kafka Connect includes schema registry which pulls in a version of guava. 

TLDR: User was using the Kafka build from the Apache website and I test with the Kafka build from the Confluent website.

**Comment by cameron** [comment]
What expedience! You beat AWS Support to the punch... I'm still stumped, but thanks for getting back to me!
